### PR TITLE
Adding conditions to CAPH

### PR DESCRIPTION
**Pull request by apluchik**
Initial commit for adding conditions to CAPH.

Also contains changes from cluster phasing pr (https://github.com/microsoft/cluster-api-provider-azurestackhci/pull/88), pending e2e testing.

### CODE REVIEW

File: cloud/scope/cluster.go [hunk 1]
```diff
@@ -160,12 +161,26 @@ func (s *ClusterScope) ListOptionsLabelSelector() client.ListOption {
...
+	conditions.SetSummary(s.AzureStackHCICluster,
+		conditions.WithConditions(
+			infrav1.NetworkInfrastructureReadyCondition,
+		),
+		conditions.WithStepCounterIfOnly(
+			infrav1.NetworkInfrastructureReadyCondition,
+		),
+	)
+
+	return s.patchHelper.Patch(context.TODO(),
```
Comment (JocelynBerrendonner) [inline cloud/scope/cluster.go:173]: would context.Background() be a better option than context.TODO() ? (see https://golang.org/pkg/context/#TODO )

File: cloud/scope/virtualmachine.go [hunk 2]
```diff
@@ -179,12 +181,27 @@ func (m *VirtualMachineScope) SetAnnotation(key, value string) {
...
+	conditions.SetSummary(m.AzureStackHCIVirtualMachine,
+		conditions.WithConditions(
+			infrav1.VMRunningCondition,
+		),
+		conditions.WithStepCounterIfOnly(
+			infrav1.VMRunningCondition,
+		),
+	)
+
+	return m.patchHelper.Patch(context.TODO(),
```
Comment (JocelynBerrendonner) [inline cloud/scope/virtualmachine.go:193]: Same question as before about context.Background()

File: cloud/scope/cluster.go (hunk 1 above)
Comment (zawachte-msft) [reply cloud/scope/cluster.go:173]: actually, the Scope Object should already have a context as part of the struct. please use that.

**Review by JocelynBerrendonner — APPROVED** [review]
Globally, this LGTM, that said I think @nwoodmsft may want to double-check.

File: cloud/scope/cluster.go (hunk 1 above)
Comment (JocelynBerrendonner) [reply cloud/scope/cluster.go:173]: Indeed! Thanks for pointing this out, Zach! That's even better!

Comment (apluchik) [reply cloud/scope/cluster.go]: Changed to use ClusterScope's context

File: cloud/scope/virtualmachine.go (hunk 2 above)
Comment (apluchik) [reply cloud/scope/virtualmachine.go]: Changed to use VirtualMachineScope's context

File: cloud/scope/cluster.go (hunk 1 above)
Comment (JocelynBerrendonner) [reply cloud/scope/cluster.go]: LGTM! Thanks!

File: cloud/scope/virtualmachine.go (hunk 2 above)
Comment (JocelynBerrendonner) [reply cloud/scope/virtualmachine.go]: LGTM! Thanks!

**Comment by zawachte-msft** [comment]
good to merge?

**Comment by apluchik** [comment]
> good to merge?

Yep, I can merge it now.

**Pull request merged**